### ISSUE TITLE
hackernews demo: Event broker TypeError on startup

### ISSUE THREAD

**Issue by oliversturm** [oliversturm/graphql-and-microservices#2]
```
➜ npm run dev

> hackernews@0.20.25 dev /home/user/git/graphql-and-microservices/demo/hackernews
> babel-node ./run.js dev

[ Client ] Time: 6110ms
Built at: 02/18/2020 4:30:55 PM
[ Server local entry point ] Time: 4671ms
Built at: 02/18/2020 4:30:53 PM
[ Local event broker entry point ] Time: 480ms
Built at: 02/18/2020 4:30:49 PM
Event broker has run into an error: TypeError: Cannot read property 'socket' of undefined
    at Object.initSockets (/home/user/git/graphql-and-microservices/demo/hackernews/node_modules/resolve-local-event-broker/lib/init-sockets.js:11:31)
    at createAndInitBroker (/home/user/git/graphql-and-microservices/demo/hackernews/node_modules/resolve-local-event-broker/lib/create-and-init-broker.js:34:14)
```

**Comment by oliversturm** [comment]
Debugging - turns out that `resolve-zeromq` dependency was missing from `package.json`. It's only strange that this didn't come up in the past. I have seen before that yarn auto-installs some deps that npm doesn't pull in, but in this case the issue comes up with both npm and yarn.

**Comment by oliversturm** [comment]
Turns out this issue is still showing on a different machine. I did a fresh `npm install`, and `node_modules/resolve-zeromq` exists, but the error comes up as before.

**Comment by oliversturm** [comment]
Right. On the main machine, I had a working installation of the modules. Now I removed `node_modules`, `package-lock.json` (and all old yarn remnants) and did `npm install` again. Result: the error comes back up.

**Comment by oliversturm** [comment]
Tried installing on top: `npm install resolve-zeromq@0.20.25` - npm changes the version in `package.json` to include a leading `^` and it pretends to do something, but the error is still there.

For posterity. Previously I debugged and found that the package `resolve-zeromq` is resolved by code in `resolve-local-event-broker/lib/index.js`, to set the property `pool/zmq` via imports (this is what the original stack trace pointed at). Since I was already suspecting an absent dependency, I checked `package.json` for `resolve-zeromq` and I didn't find it. I added it (manually) to the list, then ran `npm install` - and I was left with a working application. Since I deleted my modules again, I have been unable to reproduce that working state.

**Comment by oliversturm** [comment]
Very strange development. After a basic `npm install` (and the error coming up) I did `npm remove resolve-zeromq`, intending to reinstall afterwards. However, when I ran the application, suddenly it works! Checked: `node_modules/zeromq` is still there. Clearly this dependency is somehow pulled in from one of the other packages after all - perhaps as intended. But previously I had success after *installing* it explicitly, now I'm having success after *uninstalling* it. Still needs more tests, I guess.

**Comment by oliversturm** [comment]
I moved my `package-lock.json` to the side, removed `node_modules` again and `npm install`ed once more (`package.json` still does NOT have the `resolve-zeromq` line which was removed in the previous step). I was expecting to see that the newly generated `package-lock.json` would differ from the one I ended up with after previous steps, but that is not the case - they are both the same. However, the error is once more back.

**Comment by oliversturm** [comment]
So, here's what happens. When I run `npm install`, the package `resolve-zeromq` is pulled in as a dependency (doesn't need to be referenced specifically in `package.json`). However, it also installs its own dependency for `zeromq` in `node_modules/resolve-zeromq/node_modules/zeromq`. At this time, the version of `zeromq` installed here is `6.0.0-beta.6`, which is newer than the fixed `5.1.0` in the project's `package.json`. I don't know which part is important here - that there's an extra version (this could confuse the internal build mechanisms of resolve), that it's simply a different version, or that this version is different in some specific aspect from v5.

When I run `npm remove resolve-zeromq`, the one and only change is that the folder `node_modules/resolve-zeromq/node_modules` is deleted. As I found before, this change is not reflected in `package-lock.json`, so it is impossible to reach the same state with one standard `npm install` command for a fresh project clone.

**Comment by oliversturm** [comment]
For the time being, I have taken note of the extra installation requirements. I expect that these issues will go away with version updates in the future.
